src/Tasks/ResolveAssemblyReferences/Services/ResolveAssemblyReferenceBuildEngine.cs

```diff
@@ -0,0 +1,106 @@
+﻿using Microsoft.Build.Framework;
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Microsoft.Build.Tasks.ResolveAssemblyReferences.Services
+{
+    internal sealed class ResolveAssemblyReferenceBuildEngine : IBuildEngine7
```

Comment:
What is this for at this point?

Reply:
RAR task needs to have supplied an IBuildEngine to it. Through it communicates with internals of MSBuild. By supplying RAR task this engine I can capture all events from RAR tasks to engine and forward them to the out of proc node.